### DISCUSSION TITLE
Safe usability of using Netdata on to a personal computer

### DISCUSSION THREAD

**Discussion by ghost** [netdata/netdata#18974] [Q&A]
How safe is it to use netdata on my Windows 11 system? I'm afraid to install the agent on my computer. My system is as follows:

Processor AMD Ryzen 5 2600 Six-Core Processor 3.40 GHz
RAM 16.0 GB
System type 64-bit operating system, x64 processor
Pen and touch Pen and touch input are not available for this monitor
Edition Windows 11 Pro
Version 23H2
Installation date ‎19.‎04.‎2023
OS Build 22631.4317
Interoperability Windows Feature Experience Pack 1000.22700.1041.0

**Comment by ilyam8** [comment]
Hi, @Iliya5digital. Why are you afraid of?

**Comment by ghost** [comment]
I can break my PC after installing this agent

**Reply by ilyam8** [reply]
It will not happen.

**Comment by aldem** [comment]
@ilyam8 I would not be so sure. I installed it few hours after release, and this was the only change in my system within last couple weeks.

It was running for maybe few hours, survived overnight standby, then during the day, when I left it unattended for a while, I found that both my GPUs (one on CPU and one discrete) suddenly "disappeared" and it switched to some emergency mode (minimal resolution).

Shutdown and subsequent reboot solved the issue so it didn't look like a hardware failure and everything was fine... for a while.

Few hours later, I left laptop idling unattended again (had only browser open with couple light-weight sites), and when I got back the story repeated. Simple reboot (without power off) solved the problem - it really does not look like hardware issue.

I was very surprised and initially didn't even thought about Netdata, but taking into account that it is constantly pulling a lot of data about system state, _including_ GPU state, it _might_ cause issues due to interaction with drivers - maybe memory leaks, whatever...

Sure, I know that "after" does not imply "because of" - but lets face the facts:

- My laptop was extremely stable for the whole lifetime (~ 2.5 years) (with an exception mentioned below)
- I don't often install/reinstall stuff with the exception of updates, even those are done once per month usually (unless critical)
- Netdata installation was the _only_ change on my system before this happened
- I never experienced *any* instability before
  - Well, to be fair, 2-3 times for the whole time GPU drivers had issues but it didn't produce this effect, they recovered within seconds
- The problem arises when screen saver (blank screen) is activated and GPUs are entering power-save mode
- And finally - _after_ uninstallation of Netdata the problem disappeared.

I remember that long time ago, when I just got this laptop, I had blue screens and all sorts of instability while running some tools like HWInfo, which are fetching sensor data - but only if they were running continuously (like Netdata), so _somehow_ it could be related to (bad) interaction with drivers or APIs (memory leaks and all that stuff - not in Netdata, but in the kernel and drivers).

Obviously, my experience does not mean that similar issue will happen on other installation, it is, most likely, the specific combination of hardware and drivers, but in general, saying "it will not happen" if we are talking about anything related to IT is a bold promise - because literally _everything_ could happen, you simply can't be sure - "works for me" does not mean "works for everybody".

Just for reference, my laptop's specs:

- Lenovo Legion 5 Pro 16ACH6h, AMD GPU + NVIDIA 3060M, 64G RAM
- Windows 23H2 (exactly like OP) with all recent (couple weeks old) updates
- Nothing extraordinary installed - Visual Studio, Android Studio, few JetBrains IDEs, Steam with few games (though I play very seldom)

In the event logs I found tons of events from NVIDIA driver:

![image](https://github.com/user-attachments/assets/d824442f-4858-4332-86f7-d04fad0e5d47)

All those events were generated within few seconds - literally thousands of them. Googling didn't really help - people had similar events in completely different situations (like overclocking), and as I mentioned before it never happened to me.

As this is my primary laptop I could not afford to experiment further, but  based on the aforementioned facts I have a reason to believe that Netdata's interaction with system is the cause, though most likely not it's fault, of course.

Maybe reduction of collection interval could solve it, maybe using different APIs could help, whatever... But "out of the box" - well, it didn't go well, thus I would not recommend to "just install" unless you are prepared for surprises.

**Reply by ilyam8** [reply]
Netdata collects metrics on Win using perflib (interface provided by the system to query metrics registry), it has nothing to do with drivers. As for GPU - using external binaries like nvidia-smi (AMD only in Linux). We have Netdatas running on physical win machines and VMs and didn’t see any problems so far. I suggest you to open an issue.

**Reply by ilyam8** [reply]
Try disabling nvidia_smi collector. If continuously running nvidia_smi causes the problem… I guess we can disable this collector by default on Win.
